### PR TITLE
shim SNICallback for old and current styles

### PR DESCRIPTION
**Pull request by coolaj86**
SNICallback used to be sync, it's now been updated to be async

### CODE REVIEW

**Comment by michielbdejong** [comment]
Thanks!

**Comment by michielbdejong** [comment]
It already surprised me that SNICallback was synchronous in node at the time, nice that it's now asynchronous and that we can now support both ways.

File: examples/onthefly.js [hunk 1]
```diff
@@ -170,12 +179,18 @@ function startServer(handler, whitelist) {
...
+            // io.js always requires callback
+            shim(contexts[servername]);
             delete pendingCerts[servername];
           });
         } else {
           console.log('Rejected by whitelist function');
         }
-        return defaultContext;
+        
+        // node v0.10 mistakenly returns synchronously
```
Comment (michielbdejong) [inline examples/onthefly.js]: Do you mean node versions < 0.10 here? In which version did it change?

**Comment by coolaj86** [comment]
node <= 0.10 is sync

node v0.12 / io.js (all) are async

**Comment by coolaj86** [comment]
P.S. I didn't test my changes (bad llama), but hopefully I will as soon as I can get the basic demo to not take 89 minutes to run on my RPi....

**Pull request merged**